### PR TITLE
[Generic Database Connector] Added documentation for Microsoft SQL in developing.md

### PR DESCRIPTION
**Pull request by moxarth-elastic**
## This PR addresses the following changes:

- Add the Microsoft SQL connectors in the [developing.md](https://github.com/elastic/connectors-python/blob/main/docs/DEVELOPING.md).

#### Pre-Review Checklist
- [ ] Added a label for each target release version (example: `v7.13.2`, `v7.14.0`, `v8.0.0`)
- [x] Considered corresponding documentation changes

### CODE REVIEW

File: docs/DEVELOPING.md
Comment (wangch079) [inline docs/DEVELOPING.md:349]:
```suggestion
The [Elastic Microsoft SQL connector](../connectors/sources/mssql.py) is provided in the Elastic connectors python framework and can be used via [build a connector](https://www.elastic.co/guide/en/enterprise-search/current/build-connector.html).
```

Use relative link instead of pointing to a specific branch.

File: docs/DEVELOPING.md
Comment (wangch079) [inline docs/DEVELOPING.md:377]:
```suggestion
- The port where the Microsoft SQL Server is hosted.
- The username the connector will use to log in to the Microsoft SQL Server.
- The password the connector will use to log in to the Microsoft SQL Server.
- The database name where the connector will query data.
```

File: docs/DEVELOPING.md
Comment (wangch079) [inline docs/DEVELOPING.md:397]: IMO, `username` is a more intuitive option. ANd I see it's used for Network Drive Connector.

File: docs/DEVELOPING.md
Comment (wangch079) [inline docs/DEVELOPING.md:399]:
```suggestion
The username of the account for Microsoft SQL Server. Default value is `admin`.
```

File: docs/DEVELOPING.md
Comment (wangch079) [inline docs/DEVELOPING.md:414]:
```suggestion
Comma-separated list of tables. Default value is `*`, which indicates the connector will fetch data for all the tables. Examples:
```

File: docs/DEVELOPING.md
Comment (wangch079) [inline docs/DEVELOPING.md:421]:
```suggestion
The number of rows to fetch on each request to Microsoft SQL Server. Default value is `50`.
```

File: docs/DEVELOPING.md
Comment (wangch079) [inline docs/DEVELOPING.md:425]:
```suggestion
The number of retry attempts after failed request to Microsoft SQL Server. Default value is `3`.
```

File: docs/DEVELOPING.md
Comment (wangch079) [inline docs/DEVELOPING.md:433]:
```suggestion
Whether the connector should establish a secured connection. Default value is `False`.
```

File: docs/DEVELOPING.md
Comment (wangch079) [inline docs/DEVELOPING.md:435]:
```suggestion
ℹ️ For secured connection Microsoft SQL Driver uses the OpenSSL library, which gets a certificate from the Default Certificate Trust Store of OpenSSL.
```

File: docs/DEVELOPING.md
Comment (wangch079) [inline docs/DEVELOPING.md:439]:
```suggestion
- If the last updated time for a table is not available then all data in that table will be synced.
```

Assuming I understand the intention correctly.

File: docs/DEVELOPING.md
Comment (wangch079) [inline docs/DEVELOPING.md:452]:
```suggestion
ℹ️ The e2e test uses default values defined in [Configure Microsoft SQL connector](#configure-microsoft-sql-connector)
```

File: docs/DEVELOPING.md
Comment (wangch079) [inline docs/DEVELOPING.md:392]:
```suggestion
The port where the Microsoft SQL Server is hosted. Default value is `9090`. Examples:
```

File: docs/DEVELOPING.md
Comment (moxarth-elastic) [reply docs/DEVELOPING.md:349]: As per the previous discussion for `network drive` and `S3`. We point to a specific branch.

File: docs/DEVELOPING.md
Comment (moxarth-elastic) [reply docs/DEVELOPING.md]: Done

File: docs/DEVELOPING.md
Comment (moxarth-elastic) [reply docs/DEVELOPING.md]: To make it generic as other dialects  change to `For * in string connector will fetch data for all the Tables`.

File: docs/DEVELOPING.md
Comment (wangch079) [reply docs/DEVELOPING.md:349]: May I know the reason? This is going to be merged to `main`, and I feel it's weird to point to a file in branch 8.7. Imagine later we create a new branch 8.8, are we going to update it to 8.8 for each link in the doc or we leave it to 8.7?

File: docs/DEVELOPING.md
Comment (wangch079) [reply docs/DEVELOPING.md]: I think it's ok to leave the `key` of the configuration as `user` if it's already implemented. Just make sure to use `username` in the label and description, i.e. whatever is displayed to customers.

File: docs/DEVELOPING.md
Comment (wangch079) [reply docs/DEVELOPING.md]: Or we can change it to description (as in another PR)

https://github.com/elastic/connectors-python/blob/ae39452db0a70dc2e677d251b4d266e43bdc0513/docs/DEVELOPING.md?plain=1#L413

I feel it's easier to understand than `For * in string connector will fetch data for all the Tables`

**Review by wangch079 — APPROVED** [review]
LGTM except this https://github.com/elastic/connectors-python/pull/557#discussion_r1120141639

But I think this applies to existing connectors as well, so we can address in a separate PR.

File: docs/DEVELOPING.md
Comment (moxarth-elastic) [reply docs/DEVELOPING.md]: Yes, I have changed it `The Microsoft SQL connector will fetch data from all tables present in the configured database if the value is * .` [here](https://github.com/elastic/connectors-python/pull/557/files#diff-f8ff34865ce88864196f2848aebde961f27a4c1e8ce77313a4254dd948d32cfbR414) in same commit.

File: docs/DEVELOPING.md
Comment (moxarth-elastic) [reply docs/DEVELOPING.md]: Yes, the key is `user` But the label was `username`, So we can add `username` in the doc also.

**Comment by moxarth-elastic** [comment]
@danajuratoni Can you please review the documentation PR? So we can land it.

File: docs/DEVELOPING.md
Comment (danajuratoni) [reply docs/DEVELOPING.md:349]: Main reason is to not leave the branch with the connector client when clicking the link.
This was clarified in https://github.com/elastic/connectors-python/pull/549#discussion_r1126559525
and I'm good to use Chenhui's suggestion as that is more flexible as still keeps the users in the selected branch.

File: docs/DEVELOPING.md
Comment (danajuratoni) [reply docs/DEVELOPING.md]: The documentation is used for connector clients so we should be as exact as possible when referencing config parameters. 
Personally, I find this confusing. Are we using **keys** or **labels** in the documentation?
I don't have a strong opinion on which should be used, as long as we're consistent. I was always assuming you referenced keys in the DEVELOPING.md doc.
While most are identical, there are several parameters that use different keys.

- Keys: "host" / "port" /  "**user**" / "password" / "database" / "**tables**" / "**fetch_size**" / "**retry_count**"
- Labels: "Host" / "Port" / "**Username**" / "Password"/ "Database" / "**Comma-separated list of tables**" / "**Rows fetched per request**" / "**Retries per request**"

I see several options:
1. Rename the key for DB generic connector to `username`, that would make it consistent to other connector clients. Any reason to use "user" for DB connector while using the "username" key for Network Drive? 
2. Consistently use labels for all fields of connector client if that's the preferred approach.

My vote is for option 1.

**Comment by danajuratoni** [comment]
Other than the added comments, it looks good to me.
cc: @ppf2 @leemthompo @chriscressman - preliminary documentation for the Microsoft SQL connector client is ready for review.

File: docs/DEVELOPING.md
Comment (ppf2) [inline docs/DEVELOPING.md:360]: We should provide guidance on which specific ODBC driver versions we intend to support and which MS SQL Server editions (e.g. does it include Azure SQL as well) and versions we intend to support.

File: docs/DEVELOPING.md
Comment (ppf2) [inline docs/DEVELOPING.md:407]: This is a copy and paste error.  There's no default `xe` SQL Sever database, that's the default for Oracle Express Edition 😄 My 2 cents is to have no default for this field but make it required to be set.

File: docs/DEVELOPING.md
Comment (ppf2) [inline docs/DEVELOPING.md:421]: It can be helpful to provide some information on the tradeoff of increasing the number of rows fetched and if there's a max value.

File: docs/DEVELOPING.md
Comment (ppf2) [inline docs/DEVELOPING.md:425]: It can be helpful to provide some information on the tradeoff of increasing the # of retries and if there's a max # that can be retried or if if there's a setting for no retry limit.

File: docs/DEVELOPING.md
Comment (ppf2) [inline docs/DEVELOPING.md:429]: How does one know the expected drive name syntax to use here?  I'm assuming we are referring to the name of the driver as listed in the ODBC Datasource Administrator on Windows.  If so, perhaps we can add a link to provide guidance. 
 For example: https://learn.microsoft.com/en-us/sql/integration-services/import-export-data/connect-to-an-odbc-data-source-sql-server-import-and-export-wizard?view=sql-server-ver16#make-sure-the-driver-you-want-is-installed

Comment (ppf2) [reply docs/DEVELOPING.md:429]: Also, I will suggest using the official name for the driver.  Instead of calling it "Microsoft SQL Driver", let's refer to it as the "Microsoft ODBC Driver for SQL Server".

File: docs/DEVELOPING.md
Comment (ppf2) [inline docs/DEVELOPING.md:435]: I will suggest using the official name for the driver.  Instead of calling it "Microsoft SQL Driver", let's refer to it as the "Microsoft ODBC Driver for SQL Server".

Comment (ppf2) [reply docs/DEVELOPING.md:435]: We mentioned that the driver uses the openssl library.  Users may wonder if it requires a compatible version of the openssl library.  Perhaps we can direct them to MS on the openSSL version compatibility with their driver?

File: docs/DEVELOPING.md
Comment (ppf2) [inline docs/DEVELOPING.md:439]: I'm assuming we are referring to [sys.dm_db_index_usage_stats.last_user_update](https://learn.microsoft.com/en-us/sql/relational-databases/system-dynamic-management-views/sys-dm-db-index-usage-stats-transact-sql?view=sql-server-ver16) for the specific object_id (table) in the database_id (database). If so, it will be helpful to clarify.

File: docs/DEVELOPING.md
Comment (danajuratoni) [reply docs/DEVELOPING.md:407]: The rule of thumb should be to set default values where it makes sense, otherwise we should leave the fields empty. 

We're currently working on providing better guidance on configurable fields such as adding support for validations, tooltips or placeholders where it makes sense starting with 8.8.

File: docs/DEVELOPING.md
Comment (leemthompo) [reply docs/DEVELOPING.md]: **++** for consistency.

File: docs/DEVELOPING.md
Comment (ppf2) [reply docs/DEVELOPING.md:407]: Based on this [comment](https://github.com/elastic/connectors-python/pull/549/files#r1127785532) in the other feedback, it seems like the default value set to `xe` is required only because of E2E testing.  Perhaps we can revisit in 8.8 to see if we can decouple E2E testing dependency on requiring a default value to be set for the database field for all database connectors?

File: docs/DEVELOPING.md
Comment (moxarth-elastic) [reply docs/DEVELOPING.md]: Added relative link

File: docs/DEVELOPING.md
Comment (moxarth-elastic) [reply docs/DEVELOPING.md]: Change key user to username in #565

File: docs/DEVELOPING.md
Comment (moxarth-elastic) [reply docs/DEVELOPING.md]: Added note for privileges. Also, I added versions of MS-SQL server and ODBC driver version.

File: docs/DEVELOPING.md
Comment (moxarth-elastic) [reply docs/DEVELOPING.md:424]: same as [this](https://github.com/elastic/connectors-python/pull/549#discussion_r1127228169) query.

File: docs/DEVELOPING.md
Comment (moxarth-elastic) [reply docs/DEVELOPING.md:428]: same as [this](https://github.com/elastic/connectors-python/pull/549#discussion_r1127228169) query.

File: docs/DEVELOPING.md
Comment (moxarth-elastic) [reply docs/DEVELOPING.md]: Done, Added steps to get driver name.

File: docs/DEVELOPING.md
Comment (moxarth-elastic) [reply docs/DEVELOPING.md]: Added, the link for OpenSSL version compatibility with their driver

File: docs/DEVELOPING.md
Comment (moxarth-elastic) [reply docs/DEVELOPING.md]: Added more descriptions about [sys.dm_db_index_usage_stats.last_user_update](https://learn.microsoft.com/en-us/sql/relational-databases/system-dynamic-management-views/sys-dm-db-index-usage-stats-transact-sql?view=sql-server-ver16)

**Comment by leemthompo** [comment]
LGTM if @ppf2 is happy with the most recent updates following his requests? There were quite a few, so I want to make sure Pius is satisfied at this point. :)

**Review by leemthompo — APPROVED** [review]
Docs updates LGTM.

**Comment by moxarth-elastic** [comment]
@danajuratoni @ppf2 if there is nothing to update in the documentation PR and seems fine to you, can we get it merged? same for PR - #549 and #556

**Review by timgrein — DISMISSED** [review]

**Comment by timgrein** [comment]
buildkite test this

**Review by timgrein — APPROVED** [review]

**Comment by timgrein** [comment]
buildkite test this

**Pull request merged**
